src/app/services/web3.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import axios from 'axios';
+import Web3 from 'web3';
+import { proxyABI, ethblocksperday, ethblocksperhour } from '../web3charts/Abi'
+import { formatUnits } from "@ethersproject/units";
+
+export const web3 = new Web3(Web3.givenProvider || "wss://eth-mainnet.ws.alchemyapi.io/v2/nzZTij_2KAavafMTicQTL52SkxJf1Lkz");
+
+@Injectable({
+  providedIn: 'root'
+})
+export class Web3Service {
+
+    constracts = [];
+
+    ethCurrentBlock = null;
+
+    async init() {
+        await this.contractsInitiate()
+        // await this.getprices();
+        await this.getCurrentBlock()
+    }
+
+    private async getContractsVaults() {
+        const response = await axios.get('http://ethparser-staging.herokuapp.com/contracts/vaults');
+        if (response.status === 200) {
+            return response.data.data;
+        }
+    }
+
+    private async getCurrentBlock() {
+        const curBlock = await web3.eth.getBlockNumber()
+        this.ethCurrentBlock = curBlock;
+    }
+
+    private async contractsInitiate() {
+        const contracts = await this.getContractsVaults()
+
+        this.constracts = contracts.map(item => {
+            // @ts-ignore
+            item.web3 = new web3.eth.Contract(proxyABI, item.contract.address)
+            return item;
+        })
+    }
+
+    async getPrice<TP extends number >(
```

Comment:
okay, I think we had a misunderstanding with the Service conception
https://en.wikipedia.org/wiki/Service_layer_pattern
'price' is not web3 entity, better use another class with another name